src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -11,8 +19,28 @@ def upgrade_hw_version?(vmtype_hw_version, global_hw_version)
       vmtype_hw_version.nil? ? global_hw_version : vmtype_hw_version
     end
 
+    # NOTE : This always creates a new name
+    def get_uuid_name
```

Comment:
In general method names in Ruby beginning with `get_` should be avoided as they generally indicate that we're trying to do the wrong thing.

In this case we have a method called `get_uuid_name` that actually **sets** the `@vm_cid` instance variable. And in fact based on what I can tell every time this method is called `@vm_cid` is actually reset based on its return value.

We should just get rid of this method.